vignettes/sir_models.Rmd

```diff
@@ -0,0 +1,680 @@
+---
+title: "SIR models with odin, dust and mcstate"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{sir_models}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+```{r setup}
+library(odin.dust)
+library(mcstate)
+library(rmarkdown)
+```
+
+# Stochastic SIR model definition
+
+A simple definition of the SIR model, as given in the [odin documentation](https://mrc-ide.github.io/odin/articles/discrete.html) is:
+$$\begin{align*} 
+\frac{dS}{dt} &= -\beta \frac{SI}{N} \\
+\frac{dI}{dt} &= \beta \frac{SI}{N} - \gamma I \\
+\frac{dR}{dt} &= \gamma I \\
+\end{align*}$$
+$S$ is the number of susceptibles, $I$ is the number of infected and $R$ is the number recovered; the total population size $N = S + I + R$ is constant. $\beta$ is the infection rate, $\gamma$ is the recovery rate.
+
+Discretising this model in time steps of width $dt$ gives the following update equations for each time step:
+
+$$\begin{align*} 
+S_{t+1} &= S_t - n_{SI} \\
+I_{t+1} &= I_t + n_{SI} - n_{IR} \\
+R_{t+1} &= R_t - n_{IR}
+\end{align*}$$
+
+where
+$$\begin{align*} 
+n_{SI} &\sim B(S, 1 - e^{-\beta \frac{I}{N} \cdot dt)} \\
+n_{IR} &\sim B(I, 1 - e^{-\gamma \cdot dt)}
+\end{align*}$$
+
```

Comment:
For the *very first* example, I'd wondered about using one of the builtin examples. So we have already `dust::dust_example("sir")` which implements this model. The reasoning here is that it's less stuff to start getting started with